packages/compiler/src/expression_parser/parser.ts

```diff
@@ -189,43 +189,63 @@ export class Parser {
       input: string, location: string,
       interpolationConfig: InterpolationConfig = DEFAULT_INTERPOLATION_CONFIG): SplitInterpolation
       |null {
-    const regexp = _getInterpolateRegExp(interpolationConfig);
-    const parts = input.split(regexp);
-    if (parts.length <= 1) {
-      return null;
-    }
     const strings: string[] = [];
     const expressions: string[] = [];
     const offsets: number[] = [];
     const stringSpans: {start: number, end: number}[] = [];
     const expressionSpans: {start: number, end: number}[] = [];
-    let offset = 0;
-    for (let i = 0; i < parts.length; i++) {
-      const part: string = parts[i];
-      if (i % 2 === 0) {
-        // fixed string
+    let i = 0;
+    let inInterpolation = false;
+    let {start: interpStart, end: interpEnd} = interpolationConfig;
+    while (i < input.length) {
+      if (!inInterpolation) {
+        // parse until starting {{
+        const start = i;
+        while (i < input.length && input.substring(i, i + interpStart.length) !== interpStart) {
```

Comment:
You could use `startsWith` with an offset instead:

```suggestion
        while (i < input.length && input.startsWith(interpStart, i)) {
```

Reply:
Done